### PR TITLE
[R-package] Turn matrix to storage mode "double"

### PR DESCRIPTION
**Pull request by mayer79**
Fixes https://github.com/microsoft/LightGBM/issues/3139

Just copy-pasted code snippet from lgb.Dataset initalizer.

As I am still not very familiar with the LightGBM data logic, @Laurae2 and @jameslamb would need to confirm that this makes any sense.

### CODE REVIEW

**Comment by Laurae2** [comment]
@jameslamb This issue will also solve https://github.com/microsoft/LightGBM/issues/3094

File: R-package/R/lgb.Predictor.R [hunk 1]
```diff
@@ -142,6 +142,10 @@ Predictor <- R6::R6Class(
 
         # Check if data is a matrix
         if (is.matrix(data)) {
+        # Check whether matrix is the correct type first ("double")
+          if (storage.mode(data) != "double") {
```
Comment (jameslamb) [inline R-package/R/lgb.Predictor.R:146]:
```suggestion
          # Check whether matrix is the correct type first ("double")
          if (storage.mode(data) != "double") {
```

align comment start with the code it describes

File: R-package/tests/testthat/test_Predictor.R [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+test_that("predictions do not fail for integer input", {
+    X <- as.matrix(as.integer(iris[, "Species"]), ncol = 1L)
+    y <- iris[["Sepal.Length"]]
+    dtrain <- lgb.Dataset(X, label = y)
+    fit <- lgb.train(
+        data = dtrain
+        , objective = "regression"
+        , verbose = -1L
+    )
+    X_double <- X_integer <- X[c(1L, 51L, 101L), , drop = FALSE]
```
Comment (jameslamb) [inline R-package/tests/testthat/test_Predictor.R:12]: do you mind breaking this up into two lines please? We don't use double-`<-` anywhere else in the codebase, as I think it's a little difficult to read.

```r
X_double <- X[c(1L, 51L, 101L), , drop = FALSE]
X_integer <- X_double
```

**Comment by jameslamb** [comment]
thanks for the contribution and for using `LightGBM`, @mayer79 !

**Pull request merged**